File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    numbers = list(map(str,numbers))
    numbers.sort()
    new_numbers = sorted(numbers, key = lambda x: x*3, reverse = True)
    answer = ''.join(new_numbers) if int(''.join(new_numbers)) != 0 else str(0)
    return answer

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String[] nums = new String[numbers.length];
        for(int i = 0; i < numbers.length; i++){
            nums[i] = numbers[i] + "";
        }
        
        Arrays.sort(nums, (a, b) -> (b + a).compareTo(a + b));
        
        String answer = "";
        for(String n : nums){
            answer += n;
        }
        
        if(answer.charAt(0) == '0'){
            return "0";
        }
        
        return answer;
    }
}